4.0/en/0x21-V13-API.md

```diff
@@ -56,6 +56,18 @@ Note: Due to issues with XXE attacks against DTDs, DTD validation should not be
 | **13.4.1** | Verify that a query allow list or a combination of depth limiting and amount limiting is used to prevent GraphQL or data layer expression Denial of Service (DoS) as a result of expensive, nested queries. For more advanced scenarios, query cost analysis should be used. |  | ✓ | ✓ | 770 |
 | **13.4.2** | Verify that GraphQL or other data layer authorization logic should be implemented at the business logic layer instead of the GraphQL layer. |  | ✓ | ✓ | 285 |
 
+## V13.5 WebSocket Security Requirements
+
+| # | Description | L1 | L2 | L3 | CWE |
+| :---: | :--- | :---: | :---:| :---: | :---: |
+| **13.5.1** | Verify that only authenticated users can open WebSocket connections. | ✓ | ✓ | ✓ | 306 |
+| **13.5.2** | Verify that WebSocket Secure (wss) is used for all WebSocket connections. | ✓ | ✓ | ✓ | 319 |
+| **13.5.3** | Verify that the Origin header is checked in the initial HTTP WebSocket handshake. | ✓ | ✓ | ✓ | 346 |
+| **13.5.4** | Verify that WebSocket messages are properly sanitized and encoded server-side. | ✓ | ✓ | ✓ | 116 |
+| **13.5.5** | Verify that rate limiting is in place for WebSocket messages | ✓ | ✓ | ✓ | 770 |
+| **13.5.6** | If you are using session or channel tokens in webSockets, verify that the tokens are obtained from a response to secure POST request only, and are not sent by the server through the Websocket connection. | ✓ | ✓ | ✓ | 319 |
```

Comment:
Are you talking about initially receiving the token here, i.e. how it is produced in the first place? Please can you clarify here.

Reply:
clarified